zulipterminal/model.py

```diff
@@ -841,14 +841,17 @@ def _update_topic_index(self, stream_id: int, topic_name: str) -> None:
         Update topic order in index based on incoming message.
         Helper method called by _handle_message_event
         """
-        topic_index = self.topics_in_stream(stream_id)
-        for topic_iterator, topic in enumerate(topic_index):
+        topic_list = self.topics_in_stream(stream_id)
+        for topic_iterator, topic in enumerate(topic_list):
             if topic == topic_name:
-                topic_index.insert(0, topic_index.pop(topic_iterator))
+                topic_list.insert(0, topic_list.pop(topic_iterator))
                 return
```

Comment:
This doesn't update the data now - that's what the code says, but it took a while to figure out why. The UI still works since we update them independently. We should probably comment in the caller that it works this way, and whether we want to determine if if it's important to keep this synchronized, or how.